### PR TITLE
Fix static/proto initializers when there aren't class fields

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes https://github.com/babel/babel/pull/14334#issuecomment-1059997536
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

The helper behaviour is now strictly aligned with the transformer:

https://github.com/babel/babel/blob/0b29359f9ab1e9773aa858e79271587397013a27/packages/babel-plugin-proposal-decorators/src/transformer-2021-12.ts#L734-L740

Also added a script per https://github.com/babel/babel/pull/14334#issuecomment-1060858715. [Here](https://github.com/babel/babel/runs/5451944389?check_suite_focus=true#step:7:1) is an an example output when we have uncommitted changes after build
```
Please re-run "make build" and checkout the following changes to git
HEAD detached at pull/14335/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   packages/babel-helpers/src/helpers-generated.ts

no changes added to commit (use "git add" and/or "git commit -a")
```

/cc @wvq Please try if this PR fixes the issues you found when using decorators. Thank you!

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14335"></a>

### CODE REVIEW

File: packages/babel-helpers/src/helpers/applyDecs.js [hunk 1]
```diff
@@ -446,13 +446,19 @@ function applyMemberDecs(
       base = Class;
       metadataMap = staticMetadataMap;
       kind = kind - 5 /* STATIC */;
-
-      initializers = staticInitializers;
+      // initialize staticInitializers when we see a non-field static member
+      if (kind !== 0 /* FIELD */) {
+        staticInitializers = staticInitializers || [];
+        initializers = staticInitializers;
```
Comment (nicolo-ribaudo) [inline packages/babel-helpers/src/helpers/applyDecs.js:452]: Nit, but this probably produces a smaller output: `initializers = staticInitializers = staticInitializers || []`.

Comment (nicolo-ribaudo) [reply packages/babel-helpers/src/helpers/applyDecs.js:452]: Ah no it's ok, the minifier step already produces this code.

Comment (JLHwung) [reply packages/babel-helpers/src/helpers/applyDecs.js:452]: > initializers = staticInitializers = staticInitializers || []

This is the exact output in `helpers-generated` by terser, so I don't bother to manually minify here.

**Review by existentialism — APPROVED** [review]

**Comment by wvq** [comment]
:)  It's work's now.

**Comment by nicolo-ribaudo** [comment]
I slightly updated the PR title so that it fits in a commit message even with `(#14335)` at the end :)

**Pull request merged**